packages/example-contracts/test/CrossChainMessage.spec.ts

```diff
@@ -0,0 +1,135 @@
+      zetaTokenConsumerAddress: zetaTokenConsumerUniV2.address,
+    });
+
+    await crossChainMessageContractChainB.setInteractorByChainId(
+      chainAId,
+      ethers.utils.solidityPack(["address"], [crossChainMessageContractChainA.address])
+    );
+
+    await crossChainMessageContractChainA.setInteractorByChainId(
+      chainBId,
+      ethers.utils.solidityPack(["address"], [crossChainMessageContractChainB.address])
+    );
+  });
+
+  describe("sendHelloWorld", () => {
+    it("Should revert if the cross chain address wasn't set", async () => {
+      const unsetContract = await deployCrossChainMessageMock({
+        zetaConnectorMockAddress: zetaConnectorMockContract.address,
+        zetaTokenConsumerAddress: zetaTokenConsumerUniV2.address,
+      });
+
+      await expect(unsetContract.sendHelloWorld(chainAId)).to.be.revertedWith("InvalidDestinationChainId()");
+    });
+
+    it("Should send hello world", async () => {
+      await expect(crossChainMessageContractChainA.sendHelloWorld(chainBId, { value: parseEther("1") })).to.be.not
+        .reverted;
+    });
+  });
+
+  describe("onZetaMessage", () => {
+    it("Should revert if the caller is not the Connector contract", async () => {
+      await expect(
+        crossChainMessageContractChainA.onZetaMessage({
+          destinationAddress: crossChainMessageContractChainB.address,
+          message: encoder.encode(["address", "string"], [deployerAddress, SAMPLE_TEXT]),
+          sourceChainId: 1,
+          zetaTxSenderAddress: ethers.utils.solidityPack(["address"], [crossChainMessageContractChainA.address]),
+          zetaValueAndGas: 0,
+        })
+      ).to.be.revertedWith(`InvalidCaller("${deployer.address}")`);
+    });
+
+    it("Should revert if the cross-chain address doesn't match with the stored one", async () => {
+      await expect(
+        zetaConnectorMockContract.callOnZetaMessage(
+          ethers.utils.solidityPack(["address"], [deployerAddress]),
+          1,
+          crossChainMessageContractChainB.address,
+          0,
+          encoder.encode(["address", "string"], [zetaConnectorMockContract.address, SAMPLE_TEXT])
+        )
+      ).to.be.revertedWith("InvalidZetaMessageCall()");
+    });
+
+    describe("Given a valid message", () => {
+      it("Should emit a message", async () => {
+        const messageType = await crossChainMessageContractChainA.HELLO_WORLD_MESSAGE_TYPE();
+
+        const tx = await zetaConnectorMockContract.callOnZetaMessage(
+          ethers.utils.solidityPack(["address"], [crossChainMessageContractChainA.address]),
+          1,
+          crossChainMessageContractChainB.address,
+          0,
+          encoder.encode(["bytes32", "string"], [messageType, SAMPLE_TEXT])
+        );
+
+        const result = await tx.wait();
+        const eventNames = parseCrossChainMessageLog(result.logs);
+        expect(eventNames.filter((e) => e === "HelloWorldEvent")).to.have.lengthOf(1);
```

Comment:
Isn't queryFilter better for this? (no need of extra helpers).

Example of the Connector:

```typescript
        const zetaReceivedFilter = zetaConnectorNonEthContract.filters.ZetaReceived();
        const e1 = await zetaConnectorNonEthContract.queryFilter(zetaReceivedFilter);
        expect(e1.length).to.equal(1);
```